src/test/java/org/folio/entlinks/controller/ConsortiumAuthoritySourceFilesIT.java

```diff
@@ -0,0 +1,124 @@
+package org.folio.entlinks.controller;
+
+import static org.folio.entlinks.controller.ConsortiumLinksSuggestionsIT.COLLEGE_TENANT_ID;
+import static org.folio.entlinks.controller.ConsortiumLinksSuggestionsIT.UNIVERSITY_TENANT_ID;
+import static org.folio.support.DatabaseHelper.AUTHORITY_ARCHIVE_TABLE;
+import static org.folio.support.DatabaseHelper.AUTHORITY_SOURCE_FILE_CODE_TABLE;
+import static org.folio.support.DatabaseHelper.AUTHORITY_SOURCE_FILE_TABLE;
+import static org.folio.support.DatabaseHelper.AUTHORITY_TABLE;
+import static org.folio.support.base.TestConstants.CENTRAL_TENANT_ID;
+import static org.folio.support.base.TestConstants.authorityEndpoint;
+import static org.folio.support.base.TestConstants.authoritySourceFilesEndpoint;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import lombok.SneakyThrows;
+import org.folio.entlinks.domain.dto.AuthorityDto;
+import org.folio.entlinks.exception.AuthorityArchiveConstraintException;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.testing.extension.DatabaseCleanup;
+import org.folio.spring.testing.type.IntegrationTest;
+import org.folio.support.TestDataUtils;
+import org.folio.support.base.IntegrationTestBase;
+import org.hamcrest.Matcher;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.web.servlet.ResultMatcher;
+
+@IntegrationTest
+@DatabaseCleanup(
+  tables = {
+    AUTHORITY_TABLE,
+    AUTHORITY_ARCHIVE_TABLE,
+    AUTHORITY_SOURCE_FILE_CODE_TABLE,
+    AUTHORITY_SOURCE_FILE_TABLE},
+  tenants = {CENTRAL_TENANT_ID, COLLEGE_TENANT_ID, UNIVERSITY_TENANT_ID})
+class ConsortiumAuthoritySourceFilesIT extends IntegrationTestBase {
+
+  public static final String COLLEGE_TENANT_ID = "college";
+  public static final String UNIVERSITY_TENANT_ID = "university";
+
+  private static final String BASE_URL = "id.loc.gov/authorities/names/";
+  private static final String FULL_BASE_URL = "http://" + BASE_URL;
+  private static final String AUTHORITY_ID = "417f3355-081c-4aae-9209-ccb305f25f7e";
+  private static final String NATURAL_ID = "n12345";
+
+  @BeforeAll
+  static void prepare() {
+    setUpConsortium(CENTRAL_TENANT_ID, List.of(COLLEGE_TENANT_ID, UNIVERSITY_TENANT_ID), true);
+  }
+
+  @Test
+  @SneakyThrows
+  @DisplayName("DELETE: Should not delete authority source file with referenced authority archive in member tenant")
+  void deleteAsfWithAuthorityArchiveReference_negative_failDeletingAsf() {
```

Comment:
I think test is not clear.
Steps probably should be
1. Create source file in central tenant via API and wait until it will be propagated to member tenants
2. Create authority with reference to the source file in central tenant via API and wait until it will be propagated to member tenants
3. Delete authority in central tenant and wait until it will be propagated to member tenants
4. Try to delete source file in central tenant and verify error

Reply:
These steps is covered. The bug is next:
2. we need to create Authority in member tenant
3. then delete it in member tenant
Currently in that case we can delete it, but expected behavior - deletion must be restricted

Reply:
Improved